File: back_api/acessibilityguide/src/main/java/com/acessibilityguide/api/migrations/MigrationGenerator.java
package com.acessibilityguide.api.migrations;

import java.io.IOException;

import org.springframework.stereotype.Component;

import com.avaje.ebean.config.dbplatform.DbPlatformName;
import com.avaje.ebean.dbmigration.DbMigration;

@Component
public class MigrationGenerator {
	
	public void generate(String name, Double version) throws IOException{
		System.setProperty("ddl.migration.version", version+"");
	    System.setProperty("ddl.migration.name", name);
	    
	    DbMigration dbMigration = new DbMigration();
	    dbMigration.setPlatform(DbPlatformName.POSTGRES);
	    dbMigration.generateMigration();	    	    
	}
	
	public void generateForPendingDrops(Double version){
	    System.setProperty("ddl.migration.pendingDropsFor", version+"");

		DbMigration dbMigration = new DbMigration();
	    dbMigration.setPlatform(DbPlatformName.POSTGRES);
    	try {
			dbMigration.generateMigration();
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
	
}

File: back_api/acessibilityguide_back/src/main/kotlin/com/acessibilityguide/api/migrations/MigrationGenerator.kt
package com.acessibilityguide.api.migrations

import org.springframework.stereotype.Component
import java.io.IOException
import com.avaje.ebean.config.dbplatform.DbPlatformName
import com.avaje.ebean.dbmigration.DbMigration



/**
 * Created by eduardolopes on 15/06/17.
 */
@Component
class MigrationGenerator {

    @Throws(IOException::class)
    fun generate(name: String, version: Double?) {
        System.setProperty("ddl.migration.version", version!!.toString() + "")
        System.setProperty("ddl.migration.name", name)

        val dbMigration = DbMigration()
        dbMigration.setPlatform(DbPlatformName.POSTGRES)
        dbMigration.generateMigration()
    }

    fun generateForPendingDrops(version: Double?) {
        System.setProperty("ddl.migration.pendingDropsFor", version!!.toString() + "")

        val dbMigration = DbMigration()
        dbMigration.setPlatform(DbPlatformName.POSTGRES)
        try {
            dbMigration.generateMigration()
        } catch (e: IOException) {
            e.printStackTrace()
        }

    }


}